### ISSUE TITLE
No matrix exponential function similar to "scipy.linalg.expm"

### ISSUE THREAD

**Issue by albertomercurio** [cupy/cupy#4768]
Hello,
i noticed it is not present the matrix exponential function, similar to "scipy.linalg.expm". It is present the cupy.exp function, which calculates the exponential of each matrix element, but there is no way to get the matrix exponential widely used in physics.

```
OS                           : Linux-4.15.0-135-generic-x86_64-with-glibc2.10
CuPy Version                 : 8.4.0
NumPy Version                : 1.19.2
SciPy Version                : 1.6.1
Cython Build Version         : 0.29.22
CUDA Root                    : /home/test/anaconda3/envs/alberto_env
CUDA Build Version           : 10020
CUDA Driver Version          : 11010
CUDA Runtime Version         : 10020
cuBLAS Version               : 10202
cuFFT Version                : 10102
cuRAND Version               : 10102
cuSOLVER Version             : (10, 3, 0)
cuSPARSE Version             : 10301
NVRTC Version                : (10, 2)
Thrust Version               : 100907
CUB Build Version            : 100800
cuDNN Build Version          : 7605
cuDNN Version                : 7605
NCCL Build Version           : 2804
NCCL Runtime Version         : 2804
cuTENSOR Version             : None
Device 0 Name                : GeForce GTX 1050
Device 0 Compute Capability  : 61
```

**Comment by jogehl** [comment]
This would be a great feature for exampling for solving high dimensional partial differential equations.

**Comment by leofang** [comment]
Closed by #7744.
